docs/remote/containers-advanced.md

```diff
@@ -151,48 +151,65 @@ If you've already built the container and connected to it, run **Remote-Containe
 
 ## Avoiding extension reinstalls on container rebuild
 
-By default, VS Code will install extensions and VS Code Server inside the container's filesystem. While this has performance benefits over a locally mounted filesystem, the disadvantage is that VS Code will have to reinstall them on a container rebuild.
+By default, VS Code will install extensions and VS Code Server inside the container's filesystem. While this has performance benefits over a locally mounted filesystem, the disadvantage is that VS Code will have to reinstall them on a container rebuild. If you find yourself rebuilding frequently, you can use a local "named volume" mount so that the extensions and VS Code Server survive a container rebuild.
 
-If you find yourself rebuilding frequently, you can use a local "volume" mount so that the extensions and VS Code Server survive a container rebuild. The volume should be unique to the container since sharing the volume across multiple containers is not currently supported. To create a container volume, follow these steps:
+There are a few side effects of doing this you should be aware of:
 
-First, configure a volume mount for `~/.vscode-server` (and/or `~/.vscode-server-insiders` for VS Code Insiders). How you do this will depend on whether you specify an image, Dockerfile, or Docker Compose file in your `devcontainer.json` file.
+* Sharing the volume across multiple containers could have unintended consequences, so you should pick a unique name for each.
+* When you rebuild the container:
+  * New extensions added to devcontainer.json will **not** be automatically installed.
+  * Any `postCreateCommand` in `devcontainer.json` will **not run**.
+* Deleting the container will not automatically delete the named volume.
 
-**Dockerfile or image**:
+To create the named local volume, follow these steps:
 
-Replace `your-volume-name-goes-here` with a unique volume name for the container in `devcontainer.json` as follows:
+1. **If you are using a non-root user**, you'll need to ensure your Dockerfile creates `~/.vscode-server` and/or `~/.vscode-server-insiders` in the container.  If you do not do this, the folder will be owned by root and your connection will fail with a permissions issue. See [Adding a non-root user to your dev container](#adding-a-non-root-user-to-your-dev-container) for full details, but you can use this snippet in your Dockerfile to create the folders for a `vscode` user:
 
-```json
-"runArgs": ["-v","your-volume-name-goes-here:/root/.vscode-server"]
-```
+    ```Dockerfile
+    ARG USERNAME=vscode
+    RUN mkdir -p /home/$USERNAME/.vscode-server /home/$USERNAME/.vscode-server-insiders \
```

Comment:
Would:
```
USER someuser
RUN mkdir -p ~/.vscode-server ...
```
work and make it easier to apply?

Reply:
That would cause the commands that follow to run as the user specified.  If you've opted not to install `sudo` for security reasons, switching to the other user would be the last step.  This is also a partial of the full code that creates the user.